app/controllers/recipes_controller.rb

```diff
@@ -0,0 +1,13 @@
+class RecipesController < ApplicationController
+  def show
+    @recipe = Recipe.find(params[:id])
+    @ingredients = @recipe.ingredients
+  end
+
+  def create_ingredient
+    @recipe = Recipe.find(params[:id])
```

Comment:
The implementation of this looks great! A  more RESTful option though would be to create a separate controller for this operation, as you had to name it `create_ingredient` because a regular `create` action would be reserved for creating recipe resources. 

One RESTful approach to handle this case, is to create what we call a joins controller. In this case, you could call it the RecipeIngredientsController. In that controller, you could have a simple `create` action and in this case, it's intuitive for a developer that the purpose of that action is to create the association between a given recipe and a given ingredient.